test/unix/main.ml

```diff
@@ -107,16 +149,68 @@ let test_add t =
       if not (List.mem v l && List.mem v' l) then
         Alcotest.fail (Printf.sprintf "Adding duplicate values failed.")
 
+let test_find_present_lwt t =
+  Lwt_list.iter_p
+    (fun (k, v) ->
+      Lwt.return
+        ( match Index.find_all t k with
+        | [] ->
+            Alcotest.fail
+              (Printf.sprintf "Wrong insertion: %s key is missing." k)
+        | l ->
+            if not (List.mem v l) then
+              Alcotest.fail
+                (Printf.sprintf "Wrong insertion: %s value is missing." v) ))
+    !l
+
 let find_present_live () = test_find_present t
 
 let find_absent_live () = test_find_absent t
 
+let no_extra_values_added () = test_no_extra_values_added t
+
+let membership () = test_mem t
+
+let find_present_live_lwt () = Lwt_main.run (test_find_present_lwt t)
+
 let find_present_restart () =
   test_find_present (Index.v ~fresh:false ~log_size ~fan_out_size index_name)
 
 let find_absent_restart () =
   test_find_absent (Index.v ~fresh:false ~log_size ~fan_out_size index_name)
 
+let no_extra_values_added_restart () =
+  test_no_extra_values_added
+    (Index.v ~fresh:false ~log_size ~fan_out_size index_name)
+
+let membership_restart () =
+  test_mem (Index.v ~fresh:false ~log_size ~fan_out_size index_name)
+
+let fail_restart_not_shared () =
+  try
+    let _ =
+      Index.v ~shared:false ~fresh:true ~readonly:true ~log_size ~fan_out_size
+        index_name
+    in
+    Alcotest.fail (Printf.sprintf "Index readonly cannot be fresh.")
+  with _ -> ()
```

Comment:
I think you probably want to be using `Alcotest.check_raises` for these tests that expect an exception. It has the advantage that you can make sure you're getting the right exception.